floss/language_identifier.py

```diff
@@ -1,13 +1,65 @@
 # Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
 import os
+import re
+import mmap
+import contextlib
+from typing import Iterable
 
 import pefile
 
 import floss.logging_
+from floss.results import StaticString
+from floss.strings import extract_ascii_unicode_strings
+from floss.rust_version_database import rust_commit_hash
 
 logger = floss.logging_.getLogger(__name__)
 
 
+def identify_language(sample: str, static_strings: Iterable[StaticString]) -> str:
+    """
+    Identify the language of the binary given
+    """
+    if is_rust_bin(static_strings):
+        logger.warning("Rust Binary Detected, Rust binaries are not supported yet. Results may be inaccurate.")
+        logger.warning("Rust: Proceeding with analysis may take a long time.")
+        return "rust"
+    elif is_go_bin(sample):
+        logger.warning("Go Binary Detected, Go binaries are not supported yet. Results may be inaccurate.")
+        logger.warning("Go: Proceeding with analysis may take a long time.")
+        return "go"
+    else:
+        return "unknown"
+
+
+def is_rust_bin(static_strings: Iterable[StaticString]) -> bool:
```

Comment:
what about the other strings you've identified for potential ID, should we include them here or leave for now?

Reply:
do the commit strings show up in every one of the test files? or is there a delta between what the commit strings and the other strings cover?